### ISSUE TITLE
Time-zones should not impact DateTime (critical)

### ISSUE THREAD

**Issue by hardhub** [bridgedotnet/Bridge#3771]
It works differently on different local time-zones set in the Date/Time settings of OS.

I tested UTC+3 and UTC+1 settings.

### Steps To Reproduce

https://dev.deck.net/7c3bd36d71b3a3b54a89f3b72110c78b

```csharp
public class Program
{
    public static void Main()
    {
	var time = new DateTime(2018, 11, 04, 0, 0, 0, 0, DateTimeKind.Utc);
        
        time = time.AddDays(-1 * 7.0);
        
        Console.WriteLine(time.Kind.ToString());
            
        Console.WriteLine(time.ToString("u"));
    }
}
```

### Expected Result

```
UTC+3:
Utc
2018-10-28 00:00:00Z

UTC+1:
Utc
2018-10-28 00:00:00Z

```

### Actual Result

```
UTC+3:
Utc
2018-10-28 00:00:00Z

UTC+1:
Utc
2018-10-27 23:00:00Z

```

For me it is most critical issue ever reported here.

**Comment by hardhub** [comment]
Also it is strange that we getting the following results:

(DateTime.Now - DateTime.UtcNow).ToString();

// UTC+1: 00:59:59.9990000
// UTC+3: 02:59:59.9990000

Correct result should be: "03:00:00"

**Comment by geoffreymcgill** [comment]
The 1-millisecond difference is likely because it's taking >0.5 milliseconds to create the second DateTime object.

**Comment by hardhub** [comment]
> The 1-millisecond difference is likely because it's taking >0.5 milliseconds to create the second DateTime object.

Yes it could be.. sounds reasonable.

**Comment by hardhub** [comment]
@geoffreymcgill Any update on this?  
My application very depends on calendar and DateTime logic.  
So for now it is stuffed by workarounds for this issue.

**Comment by hardhub** [comment]
No reply more than 2 months?

**Comment by geoffreymcgill** [comment]
I believe we have this issue fixed. You can test your original Deck.NET sample in our **dev** instance:

https://dev.deck.net/7c3bd36d71b3a3b54a89f3b72110c78b

We still have a lot more testing to do, but if everything goes well, this fix will be included in the upcoming Bridge 17.8.0 release.

**Comment by geoffreymcgill** [comment]
This scenario has been fixed. After another review and final testing, the fix will be included in the upcoming Bridge 17.8.0 release.

You can confirm with the following dev.deck.net sample:

https://dev.deck.net/7c3bd36d71b3a3b54a89f3b72110c78b

**Comment by fabriciomurta** [comment]
A full test over all time zones have been undegone and all looks good, so the fix for this will be available next Bridge release!
